app/Http/Controllers/Webapp/PostController.php

```diff
@@ -0,0 +1,1890 @@
+                // Insert new post-keyword relationship if not exist
+                $postMetaSkw = new PostMeta();
+                $postMetaSkw->post_id = $post->id;
+                $postMetaSkw->keyword_id = $subKeyword->id;
+                $postMetaSkw->save();
+            }
+        }
+
+        return [
+            'data' => [
+                'id' => $post->id,
+                'checksum' => $calculatedChecksum
+            ],
+            'next' => action('Webapp\\PostController@getEdit', $post->id)
+        ];
+    }
+
+    public function getPublish(Request $request, $postId = 0) {
+        $this->activityLog->setPostId($postId);
+
+        if(empty($post = Post::find($postId))
+            || $this->currentUser->is(User::CONTRIBUTOR)) {
+            $this->activityLog->setInfo('Not found');
+            return '404 Not Found';
+        }
+
+        $this->activityLog->addMeta('PreviousPublishedStatus', $post->published_status);
+
+        if(!$post->canPublishBy($this->currentUser)) {
+            $this->_flashMessage($request, 'Publish is not permitted', 'danger');
+            $this->activityLog->setInfo('Not permitted');
+            return redirect()->to(action('Webapp\\PostController@getIndex'))->send();
+        }
+
+        if($post->isLock($this->currentSession)) {
+            $this->_flashMessage($request, 'このコンテンツは現在ロックされています。変更できません。', 'danger');
+            $this->activityLog->setInfo('Locked');
+            if(!$request->get('selected_ids'))
+                return redirect()->to(action('Webapp\\PostController@getIndex'))->send();
+            else
+                return;
+        }
+
+        if($post->published_status == 1) {
+            $this->_flashMessage($request, '更新しました。', 'warning');
+            $this->activityLog->setInfo('Already published');
+            if(!$request->get('selected_ids'))
+                return redirect()->to(action('Webapp\\PostController@getIndex'))->send();
+            else
+                return;
+        }
+
+        $data = array(
+            'site_id' => $post->site_id,
+            'title' => $post->title,
+            'desc' => $post->description,
+            'thumb' => $post->feature_img,
+            'mainKey' => empty(PostMeta::getMainKeywordByPostId($postId)) ? '' : $postId
+        );
+
+        $mess_error = Post::checkPostDataRequireField($data, true);
+
+        if($mess_error != ''){
+            $this->_flashMessage($request, $mess_error, 'danger');
+            $this->activityLog->setInfo('Missing required field');
+            return redirect()->to(action('Webapp\\PostController@getEdit', $postId))->send();
+        }
+
+        if($post->site_id == 0) {
+            $this->_flashMessage($request, 'ターゲットサイトを選んでください', 'warning');
+            $this->activityLog->setInfo('Missing target site');
+            if(!$request->get('selected_ids'))
+                return redirect()->to(action('Webapp\\PostController@getEdit', $postId))->send();
+            else
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.relipa.blog/coding_rules#rule_key=php%3AS121)